### PR TITLE
HomeWork10

### PR DESCRIPTION
**Pull request by DadizhaO**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (SerhiiPasko) [inline pom.xml:2]: Why I read code that you write on lesson?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    <dependencies>
+        <!-- https://mvnrepository.com/artifact/junit/junit -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
```
Comment (SerhiiPasko) [inline pom.xml:66]: What is?

File: src/test/java/Lesson10/FruitShopUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testAllFreshAndFruitTypeNullPointer() {
+        exception.expect(NullPointerException.class);
+        fruitShop.allFreshAndFruitType(APPLE, null);
+    }
+
+    @Test
```
Comment (SerhiiPasko) [inline src/test/java/Lesson10/FruitShopUnitTest.java:129]: Good

File: src/test/java/Lesson10/FruitShopUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(JUnitParamsRunner.class)
+public class FruitShopUnitTest {
+
+    private static Fruit apple;
+    private static Fruit pear;
+    private static Fruit strawberry;
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
```
Comment (SerhiiPasko) [inline src/test/java/Lesson10/FruitShopUnitTest.java:25]: Why do you use this rule?
You can't test situations with throws exceptions

**Review by SerhiiPasko — APPROVED** [review]
Read comments

File: src/test/java/Lesson10/FruitShopUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        List<Fruit> expected = fruitShop.getFruits();
+        Assert.assertNotNull(expected);
+    }
+
+
+    @Test
+    public void testAllFresh() {
+
+        List<Fruit> exp = new ArrayList<>();
+        exp.add(apple);
```
Comment (Cherkashyn) [inline src/test/java/Lesson10/FruitShopUnitTest.java:67]: Variable names

File: src/test/java/Lesson10/FruitShopUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    @Test
+    public void testAllFreshNothing() {
+        List<Fruit> actual = fruitShop.allFresh(LocalDate.of(2099, Month.APRIL, 2));
+        assertEquals(0, actual.size());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testAllFreshDateNull() {
+        fruitShop.allFresh(null);
```
Comment (Cherkashyn) [inline src/test/java/Lesson10/FruitShopUnitTest.java:81]: You should add here a method allFresh invocation, with parameters, not a single null.

File: src/test/java/Lesson10/FruitShopUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    @After
+    public void afterTest() {
+        fruitShop = null;
+    }
+
+
+    @Test
+    public void testGetFruits() {
+        List<Fruit> expected = fruitShop.getFruits();
+        Assert.assertNotNull(expected);
```
Comment (Cherkashyn) [inline src/test/java/Lesson10/FruitShopUnitTest.java:59]: Use static imports. It's easier

**Review by Cherkashyn — APPROVED** [review]

**Pull request merged**